db/hybrid_logical_vector.go

```diff
@@ -157,3 +169,87 @@ func (hlv *HybridLogicalVector) GetVersion(sourceID string) uint64 {
 	}
 	return latestVersion
 }
+
+func (hlv *HybridLogicalVector) MarshalJSON() ([]byte, error) {
+
+	persistedHLV, err := hlv.convertHLVToPersistedFormat()
+	if err != nil {
+		return nil, err
+	}
+
+	return base.JSONMarshal(*persistedHLV)
+}
+
+func (hlv *HybridLogicalVector) UnmarshalJSON(inputjson []byte) error {
+	persistedJSON := PersistedVersionVector{}
+	err := base.JSONUnmarshal(inputjson, &persistedJSON)
+	if err != nil {
+		return err
+	}
+	// convert the data to in memory format
+	hlv.convertPersistedHLVToInMemoryHLV(persistedJSON)
+	return nil
+}
+
+func (hlv *HybridLogicalVector) convertHLVToPersistedFormat() (*PersistedVersionVector, error) {
+	persistedHLV := PersistedVersionVector{}
+	cvCasByteArray := base.Uint64CASToLittleEndianHex(hlv.CurrentVersionCAS)
```

Comment:
Since cvCas is omitempty, do we want to do a zero check here first before converting to a non-empty string?